src/VisualStudio/Core/Def/Implementation/NavigationBar/NavigationBarClient.cs

```diff
@@ -277,24 +275,27 @@ int IVsDropdownBarClient3.GetAutomationProperties(int iCombo, out string pbstrNa
         }
 
         int IVsDropdownBarClient3.GetEntryImage(int iCombo, int iIndex, out int piImageIndex, out IntPtr phImageList)
+        {
+            // This class implements IVsDropdownBarClient4 and expects IVsDropdownBarClient4.GetEntryImage() to be called instead
+            phImageList = IntPtr.Zero;
+            piImageIndex = -1;
+
+            return VSConstants.E_UNEXPECTED;
+        }
+
+        ImageMoniker IVsDropdownBarClient4.GetEntryImage(int iCombo, int iIndex)
         {
             var item = GetItem(iCombo, iIndex);
 
-            // If this is a project item, try to get the actual proper image from the VSHierarchy it 
-            // represents.  That way the icon will always look right no matter which type of project
-            // it is.  For example, if phone/Windows projects have different icons, then this can 
-            // ensure we get the right icon, and not just a hard-coded C#/VB icon.
             if (item is NavigationBarProjectItem projectItem)
             {
-                if (_workspace.TryGetImageListAndIndex(_imageService, projectItem.DocumentId.ProjectId, out phImageList, out piImageIndex))
```

Comment:
do you still need to keep that implementation because of Object Browser and such, which still need imagelist?

Reply:
It seems the remaining use is the "preview changes" dialog for us previewing a codefix.

Reply:
(and it's possible that can move too, but that's not today's problem.)

Reply:
who owns it? if the editor, please file a bug for tracking

Reply:
I have no idea who owns it, but I'll follow up.